contracts/carbon/Strategies.sol

```diff
@@ -514,17 +522,23 @@ abstract contract Strategies is Initializable {
     /**
      * @dev calculates the required amount plus fee
      */
-    function _addFee(uint128 amount) private view returns (uint128) {
+    function _addFee(uint128 amount, uint128 pairId) private view returns (uint128) {
+        // override protocol-wide trading fee with custom one if it's set for the pair
+        uint32 tradingFeePPMOverrides = _tradingFeePPMOverrides[pairId];
```

Comment:
Thinking out loud: what about encoding (e.g., in a bit) in the pair ID that it has a custom fee? This way, we would know to avoid a storage read for every pair that does not.

Reply:
I think that's a bit too complicated an optimization to save a storage read.
Regardless, I'm open for a discussion if we'd want to change it.